src/main/java/by/dragonsurvivalteam/dragonsurvival/common/handlers/DragonBonusHandler.java

```diff
@@ -137,4 +110,44 @@ public static void addFireProtectionToCaveDragonDrops(BlockDropsEvent dropsEvent
             dropsEvent.getDrops().forEach(drop -> drop.getData(DragonSurvival.ENTITY_HANDLER).isFireImmune = true);
         }
     }
+
+    // We *need* this event to trigger last so that if other mods cancel the event, we don't set the player as jumping
+    @SubscribeEvent(priority = EventPriority.LOWEST)
+    public static void flagPlayersAsJumping(PlayerTickEvent.LivingJumpEvent event) {
+        if (event.getEntity() instanceof Player player) {
+            DragonStateHandler handler = DragonStateProvider.getData(player);
+            if (handler.isDragon()) {
+                handler.isJumping = true;
+            }
+        }
+    }
+
+    // TODO: This could potentially be more efficient by putting some hook where the player gets detected as being on the ground
+    @SubscribeEvent
+    public static void flagPlayersAsNotJumping(PlayerTickEvent.Post event) {
+        DragonStateHandler handler = DragonStateProvider.getData(event.getEntity());
+        if(handler.isDragon() && event.getEntity().onGround()) {
```

Comment:
If someone jumps from somewhere and initiates some sort of flight we might be stuck in a "jump" mode
We can just remove the jump flag after doing the logic in the `LivingFallEvent`

Since "falling" means "player hit the ground" (it's added in the `LivingEntity#causeFallDamage` method)
(Which is called from `Block#fallOn`)

(Though with that event we would need to receive cancelled in order to always clear the flag)